pay-api/src/pay_api/services/statement.py

```diff
@@ -358,8 +358,9 @@ def _populate_statement_summary(cls, statement: StatementModel, statement_invoic
 
         latest_payment_date = None
         for invoice in statement_invoices:
-            if latest_payment_date is None or invoice.payment_date > latest_payment_date:
-                latest_payment_date = invoice.payment_date
+            if invoice.payment_date is not None:
```

Comment:
Less nesting if just continue if invoice.payment_date isn't there

if Invoice.payment_date is None:
   continue

Reply:
```
latest_payment_date = max(
    (invoice.payment_date for invoice in statement_invoices if invoice.payment_date is not None),
    default=None
)
```